### PR TITLE
Feature/add content downloading

### PR DESCRIPTION
**Pull request by kdwils**
Added functionality to pass a chosen release to a download client based on the download client id provided in the request. There is probably still more to add here config wise for the download clients which would flesh out the rest of the columns in the download client table.

It was pretty slow though. I am thinking that once we have some sort of state for the media tracked by the manager this flow will change quite a bit. 

My thoughts here are that we could a new piece of media with some sort of state to track, and then the manager would attempt to get that media to the desired state. The more expensive operations such as querying indexers and passing along the release to a download client would happen at a later time.

I was able to send a torrent to transmission by making the following request

```
curl -X POST 'http://localhost:8080/api/v1/library/movies' \
  --header 'Content-Type: application/json' \
  --data-raw $'{
  "tmdbID": 917496,
  "qualityProfileID": 3,
  "downloadClientID": 2
}'
```

### CODE REVIEW

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -237,8 +240,26 @@ func (m MediaManager) AddMovieToLibrary(ctx context.Context, request AddMovieReq
 
 	slices.SortFunc(releases, sortReleaseFunc())
 	chosenRelease := releases[len(releases)-1]
+
 	log.Info("found release", "title", chosenRelease.Title)
-	return chosenRelease, nil
+
+	downloadClient, err := m.GetDownloadClient(ctx, int64(request.DownloadClientID)) // yolo cast?
```
Comment (kasuboski) [inline pkg/manager/manager.go:246]: Which download client to use is going to depend on the chosen release right? Like torrent vs usenet. Can be a future problem

File: pkg/storage/sqlite/sqlite.go [hunk 2]
```diff
@@ -318,3 +316,41 @@ func (s SQLite) handleStatement(ctx context.Context, stmt sqlite.Statement) (sql
 
 	return result, tx.Commit()
 }
+
+// GetQualityProfileItem gets a quality item that belongs to a profile
```
Comment (kasuboski) [inline pkg/storage/sqlite/sqlite.go:320]: The comments don't match

File: pkg/storage/sqlite/sqlite.go [hunk 3]
```diff
@@ -318,3 +316,41 @@ func (s SQLite) handleStatement(ctx context.Context, stmt sqlite.Statement) (sql
...
+func (s SQLite) ListDownloadClients(ctx context.Context) ([]*model.DownloadClient, error) {
+	items := make([]*model.DownloadClient, 0)
+	stmt := table.Indexer.SELECT(table.DownloadClient.AllColumns).FROM(table.DownloadClient).ORDER_BY(table.DownloadClient.ID.ASC())
+	err := stmt.QueryContext(ctx, s.db, &items)
+	return items, err
+}
+
+// DeleteQualityDefinition deletes a quality
+func (s SQLite) DeleteDownloadClient(ctx context.Context, id int64) error {
+	stmt := table.DownloadClient.DELETE().WHERE(table.DownloadClient.ID.EQ(sqlite.Int64(id))).RETURNING(table.DownloadClient.AllColumns)
```
Comment (kasuboski) [inline pkg/storage/sqlite/sqlite.go:338]: Does this need a returning clause if we don't return anything? I think I did that somewhere too and had questioned it

File: pkg/manager/manager.go (hunk 1 above)
Comment (kdwils) [reply pkg/manager/manager.go:246]: I did add a type on the schema so we'll be able to query on something like torrent vs usenet, just wasn't sure how this wa going to look

File: pkg/storage/sqlite/sqlite.go (hunk 3 above)
Comment (kdwils) [reply pkg/storage/sqlite/sqlite.go:338]: I don't think its needed

File: pkg/storage/sqlite/sqlite.go (hunk 2 above)
Comment (kdwils) [reply pkg/storage/sqlite/sqlite.go]: fixed all of these

**Review by kasuboski — APPROVED** [review]

**Pull request merged**